encoders/nlp/TransformerTorchEncoder/__init__.py

```diff
@@ -8,170 +8,156 @@
 from jina.executors.decorators import batching, as_ndarray
 from jina.executors.devices import TorchDevice
 from jina.executors.encoders import BaseEncoder
-from jina.executors.encoders.helper import reduce_mean, reduce_max, reduce_min, reduce_cls
 from jina.helper import cached_property
 from jina.logging import default_logger
 
 
-def auto_reduce(model_outputs: 'np.ndarray', mask_2d: 'np.ndarray', model_name: str) -> 'np.ndarray':
-    """
-    Automatically creates a sentence embedding from its token embeddings.
-        * For BERT-like models (BERT, RoBERTa, DistillBERT, Electra ...) uses embedding of first token
-        * For XLM and XLNet models uses embedding of last token
-        * Assumes that other models are language-model like and uses embedding of last token
-    """
-    if 'bert' in model_name or 'electra' in model_name:
-        return reduce_cls(model_outputs, mask_2d)
-    if 'xlnet' in model_name:
-        return reduce_cls(model_outputs, mask_2d, cls_pos='tail')
-    default_logger.warning('Using embedding of a last token as a sequence embedding. '
-                           'If that is not desirable, change `pooling_strategy`')
-    return reduce_cls(model_outputs, mask_2d, cls_pos='tail')
-
-
 class TransformerTorchEncoder(TorchDevice, BaseEncoder):
     """
-    Internally, TransformerTorchEncoder wraps the tensorflow-version of transformers from huggingface.
+    Internally, TransformerTorchEncoder wraps the pytorch version of transformers from huggingface.
     """
 
     def __init__(
         self,
         pretrained_model_name_or_path: str = 'sentence-transformers/distilbert-base-nli-stsb-mean-tokens',
+        base_model: str = None,
         pooling_strategy: str = 'mean',
         layer_index: int = -1,
         max_length: Optional[int] = None,
-        truncation_strategy: str = 'longest_first',
+        truncation_strategy: bool = True,
```

Comment:
`strategy` seems a weird name for a boolean option right?

Reply:
True, I can probably just leave it at `'longest_first'`, as it's deprecated anyways